app/controllers/users/two_factor_authentication_setup_controller.rb

```diff
@@ -14,22 +14,24 @@ def index
     end
 
     def create
-      result = submit_form
-      analytics.user_registration_2fa_setup(**result.to_h)
-
-      if result.success?
-        process_valid_form
-      elsif (result.errors[:selection].include? 'phone') ||
-            IdentityConfig.store.kantara_2fa_phone_restricted
-        flash[:phone_error] = t('errors.two_factor_auth_setup.must_select_additional_option')
-        redirect_to authentication_methods_setup_path(anchor: 'select_phone')
+      if params[:two_factor_options_form].blank?
```

Comment:
There's a lot of conditional branching now. Maybe the code in this condition could be extracted to a method so that the condition can be written in one line, such as:

```
redirect_to_whatever if params[:two_factor_options_form].blank?
```

Or another pattern could be found.

Reply:
Or rather something like `return redirect_to_whatever ...` or a different pattern =)

In an effort to reduce the if/else branching.